addon/services/user-agent.js

```diff
@@ -1,106 +1,135 @@
-    const parser = get(this, 'UAParser');
-
-    const browser = parser.getBrowser();
-    const device = parser.getDevice();
-    const engine = parser.getEngine();
-    const os = parser.getOS();
-
-    setProperties(this, assign({
-
-      browser: {
-        info: browser,
-        isChrome: isEqual(browser.name, 'Chrome'),
-        isChromeHeadless: isEqual(browser.name, 'Chrome Headless'),
-        isEdge: isEqual(browser.name, 'Edge'),
-        isFirefox: isEqual(browser.name, 'Firefox'),
-        isIE: isEqual(browser.name, 'IE') ||
-              isEqual(browser.name, 'IE Mobile'),
-        isSafari: isEqual(browser.name, 'Safari') ||
-                  isEqual(browser.name, 'Mobile Safari')
-      },
-
-      device: {
-        info: device,
-        isConsole: isEqual(device.type, 'console'),
-        isDesktop: !device.type,
-        isMobile: isEqual(device.type, 'mobile'),
-        isTablet: isEqual(device.type, 'tablet')
-      },
-
-      engine: {
-        info: engine,
-        isWebkit: isEqual(engine.name, 'WebKit')
-      },
-
-      os: {
-        info: os,
-        isAndroid: isEqual(os.name, 'Android'),
-        isIOS: isEqual(os.name, 'iOS'),
-        isLinux: [
-          'CentOS', 'Fedora', 'Linpus', 'Linux', 'MeeGo',
-          'PCLinuxOS', 'RedHat', 'SUSE', 'Ubuntu', 'VectorLinux'
-        ].indexOf(os.name) > -1,
-        isMacOS: isEqual(os.name, 'Mac OS'),
-        isWindows: [
-          'Windows', 'Windows Phone', 'Windows Mobile'
-        ].indexOf(os.name) > -1
-      }
-
-    }, parser));
+  device: computed('_parser', function() {
+    const device = get(this, '_parser').getDevice();
 
+    return {
+      info: device,
+      isConsole: device.type === 'console',
+      isDesktop: !device.type,
+      isMobile: device.type === 'mobile',
+      isTablet: device.type === 'tablet'
+    };
   }),
 
-  init() {
-    this._super(...arguments);
-    this.setupService();
-  }
+  engine: computed('_parser', function() {
+    const engine = get(this, '_parser').getEngine();
+
+    return {
+      info: engine,
+      isWebkit: engine.name === 'WebKit'
+    };
+  }),
+
+  os: computed('_parser', function() {
+    const os = get(this, '_parser').getOS();
+
+    return {
+      info: os,
+      isAndroid: os.name === 'Android',
+      isIOS: os.name === 'iOS',
+      isLinux: [
+        'CentOS', 'Fedora', 'Linpus', 'Linux', 'MeeGo',
+        'PCLinuxOS', 'RedHat', 'SUSE', 'Ubuntu', 'VectorLinux'
+      ].indexOf(os.name) > -1,
+      isMacOS: os.name === 'Mac OS',
+      isWindows: [
+        'Windows', 'Windows Phone', 'Windows Mobile'
+      ].indexOf(os.name) > -1
+    };
+  }),
 
+  setupService() {
+    deprecate(
+      'Usage of the private method `setupService` is deprecated. To force an update, set the `userAgent` property.',
+      false,
+      { id: 'ember-useragent.service.setupService', until: '1.0.0' }
+    );
+
+    this.notifyPropertyChange('userAgent');
```

Comment:
This method should not have been used by users directly in the first place, but some might have, so I added another deprecation 😝